builder/azure/arm/resource_resolver.go

```diff
@@ -75,41 +76,34 @@ func getResourceGroupNameFromId(id string) string {
 	return xs[4]
 }
 
-func findManagedImageByName(client *AzureClient, name, resourceGroupName string) (*compute.Image, error) {
-	images, err := client.ImagesClient.ListByResourceGroupComplete(context.TODO(), resourceGroupName)
+func findManagedImageByName(client *AzureClient, name, subscriptionId, resourceGroupName string) (*hashiImagesSDK.Image, error) {
+	id := commonids.NewResourceGroupID(subscriptionId, resourceGroupName)
+	images, err := client.ImagesClient.ListByResourceGroupComplete(context.TODO(), id)
```

Comment:
Not really part of the review but I'm curious why the code has context.TODO() as opposed to reusing the ctx created during runtime? Guessing there is no context being passed in would be my first guess.

Reply:
Yes! So the context for the clients is mostly used for canceling and polling deadlines, I have not passed it through to here yet since it was only one get request used outside of a step, happy to add a real context here if you think it'd be better